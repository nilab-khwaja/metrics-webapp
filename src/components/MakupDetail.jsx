import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import '../styles/makeupdetail.css';

const MakupDetail = () => {
  const { produtId } = useParams();
  const makeup = useSelector((state) => state.makeups.makeups
    .find((item) => item.id === parseInt(produtId, 10)));

  if (!makeup) {
    return <div>Product not found</div>;
  }

  return (
    <div className="makeup-detail">
      <h1>{makeup.name}</h1>
      <div className="detail">
        <div className="image"><img src={makeup.image_link} alt={makeup.name} /></div>
        <ul className="description">
          <li>
            <h3>
              <strong>Product type: </strong>
              {makeup.product_type}
            </h3>
          </li>
          <li>
            <p>
              <strong> Price:  </strong>
              $
              {makeup.price}
              /piece
            </p>
          </li>
          <li>
            <p>
              <strong> Price-sign: </strong>
              $
              {makeup.price_sign}
              /piece
            </p>
          </li>
          <li>
            <p>
              <strong> Description: </strong>
              {makeup.description}
            </p>
          </li>
          <li>
            <h3> Colors: </h3>
            <ul>
              {makeup.product_colors.map((color) => (
                <li key={color.hex_value} className="colors">
                  <div style={{
                    backgroundColor: color.hex_value, width: '20px', height: '20px', display: 'inline-block',
                  }}
                  />
                  {color.colour_name}
                </li>
              ))}
            </ul>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default MakupDetail;
