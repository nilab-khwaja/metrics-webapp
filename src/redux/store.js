import { configureStore } from "@reduxjs/toolkit";  
import makeupReducer from "./makeupSlice";

const store = configureStore({
    reducer: {
        makeups:makeupReducer,
    }, 
});

export default store