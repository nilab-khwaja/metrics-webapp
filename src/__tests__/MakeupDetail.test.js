import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import MakupDetail from '../components/MakupDetail';
import store from '../redux/store';

describe('MakupDetail component', () => {
  it('should render correctly', () => {
    const makeup = {
      id: 1,
      name: 'Product Name',
      product_type: 'Type',
      price: 10,
      price_sign: '$',
      description: 'Product Description',
      product_colors: [
        { hex_value: '#ffffff', colour_name: 'White' },
        { hex_value: '#000000', colour_name: 'Black' },
      ],
      image_link: 'https://example.com/image.png',
    };

    const { container } = render(
      <Provider store={store}>
        <MakupDetail makeup={makeup} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
