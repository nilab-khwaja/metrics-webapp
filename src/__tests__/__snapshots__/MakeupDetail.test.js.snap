// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakupDetail component should render correctly 1`] = `
<div>
  <div>
    Product not found
  </div>
</div>
`;
